clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -200,7 +200,7 @@ public synchronized boolean containsTopic(String topic) {
      * Updates the cluster metadata. If topic expiry is enabled, expiry time
      * is set for topics if required and expired topics are removed from the metadata.
      */
-    public synchronized void update(Cluster cluster, long now) {
+    public synchronized void update(Cluster cluster, boolean hasUnavailablePartitions, long now) {
```

Comment:
It's a fair point that `Cluster` is public and we should be careful about what we add there.